sbncode/TPCReco/NuVertexChargeTree_module.cc

```diff
@@ -706,11 +936,74 @@ void sbn::NuVertexChargeTree::FillNeutrino(const simb::MCTruth &nu,
     // see if there is a match
     //
     // Total up the energy fromthis particle
-    std::vector<const sim::IDE*> particle_ides(backtracker->TrackIdToSimIDEs_Ps(G4->TrackId()));
     float thisVisE = 0.;
-    for (const sim::IDE *ide: particle_ides) thisVisE += ide->energy;
+    float thisQ = 0.;
+
+    std::map<geo::WireID, float> chargemap;
```

Comment:
It may simplify the way chargemap is currently used. But I also build that map to enable other possible development of truth based variables. So changing the way chargemap works would make future development more difficult, without an appreciable performance gain.